### ISSUE TITLE
[next]between returns a SELECTION<ARRAY> instead of TABLE_SLICE if boundaries are in the wrong order

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#3689]
On 1.16.x

```
r.db('017e8520549f2037ecd618af255a8364').table('6eb6699f38d3bf5548d5007336bd2eeb').between(3, 2, {index: "id"}).typeOf()
// returns
"SELECTION<ARRAY>"

r.db('017e8520549f2037ecd618af255a8364').table('6eb6699f38d3bf5548d5007336bd2eeb').between(1, 2, {index: "id"}).typeOf()
"TABLE_SLICE"
```

It only broke some of my backtraces tests. I can still chain an `orderBy` with an index in both cases, so I guess it's not that bad?

**Comment by danielmewes** [comment]
Thanks for the report.
Is the reverse one even working? (@mlucy)

Not holding the release for this, and I'm not sure if we should fix it in a point release either since it's theoretically breaking compatiblity. Scheduling for 2.0 for now.

**Comment by mlucy** [comment]
That's a bug, but it probably isn't a hugely problematic one, especially since ordering by an index afterward seems to still work.  We should fix this for 2.0 but I agree it isn't worth holding the release for.

**Comment by VeXocide** [comment]
In CR 2740 by @mlucy

**Comment by VeXocide** [comment]
Merged into `next` via ccdc975d2a5f26878ca478bd65d7b17bb6e4d30d and b464c4e3527054c3fea23d5ff6c836e4361ee255, and cherry-picked into `v2.0.x` via f4846772b383b2786a079bff37d924585f9919ac and 78eca67af7b2003f00d7f30b829828e24b299bfe respectively.
